### ISSUE TITLE
Meaning of a "session"

### ISSUE THREAD

**Issue by HansBambel** [khaleelasyraaf/Mobius_Client#8]
Maybe instead of recording for 8 hours (or something like that) users can upload a walk, bike or a car session. This way we have cleaner data and do less unnecessary recording.

So, basically, users go for a walk, start the session, end the session, then upload it already.

What do you think?

**Comment by dimstudio** [comment]
I am not completely sure.. we need the unnecessary recordings as counterexamples I believe later on to recognise a walking action right?

**Comment by khaleelasyraaf** [comment]
I agree with Kevin. We'll have smaller files too. 

Or the user can start and end the session only when they're going out, uploads them by the end of the day, as long as the user stays logged in and did not close the app.

**Comment by HansBambel** [comment]
> I am not completely sure.. we need the unnecessary recordings as counterexamples I believe later on to recognise a walking action right?

Counterexample for what though? Isn't the idea to give the most likely transportation mode? When it is framed like this it should not be a problem, because no movement is no transportation anyway, right?

> I agree with Kevin. We'll have smaller files too.
> 
> Or the user can start and end the session only when they're going out, uploads them by the end of the day, as long as the user stays logged in and did not close the app.

If we do the latter, then we need to change that the sessions get overwritten after pressing "start". The zipping is still all files in the data folder anyway.

**Comment by khaleelasyraaf** [comment]
> If we do the latter, then we need to change that the sessions get overwritten after pressing "start". The zipping is still all files in the data folder anyway.

I don't think we should overwrite the previous session because the new session won't do that after pressing "start". We can stop a session (let's say) now, start a new one in an hour and still have the data recorded in the same files. Unless the user keys in a new ID, then we'll have new files.

**Comment by HansBambel** [comment]
> > If we do the latter, then we need to change that the sessions get overwritten after pressing "start". The zipping is still all files in the data folder anyway.
> 
> I don't think we should overwrite the previous session because the new session won't do that after pressing "start". We can stop a session (let's say) now, start a new one in an hour and still have the data recorded in the same files. Unless the user keys in a new ID, then we'll have new files.

I mixed it up. I thought right now we overwrite the session, but we append the new readings, right? So everything should be fine.

**Comment by khaleelasyraaf** [comment]
> I mixed it up. I thought right now we overwrite the session, but we append the new readings, right? So everything should be fine.

Yes exactly ;) your suggestion works too, but since the users need an internet connection to upload the files so I think by doing it at the end of the day would be better as they'll be at home anyway.

**Comment by dimstudio** [comment]
> Counterexample for what though? Isn't the idea to give the most likely transportation mode? When it is framed like this it should not be a problem, because no movement is no transportation anyway, right?

But "no movement" isn't it also a class? 

In general, I am fine with the solution of the recording on-demand. However, I would not replace what we have been doing so far, but try to keep the option of recording in the background

**Comment by HansBambel** [comment]
I think there is a misunderstanding. What I mean with the on-demand recording is that it is still able to record in the background, but instead of recording for 8 hours (where the user maybe only was biking for 20 minutes) we have a recording of mostly the transportation method/s instead. 

For now, we didn't have a "no movement" class, but of course we can simply create one if none of the other modes are selected.

**Comment by dimstudio** [comment]
Yeah, it is still no clear for me. If the app records data only when the transportation mode is selected, doesn't that also mean we are not going to have those moments of "no movement"?

**Comment by HansBambel** [comment]
Wait, the user is still only clicking on "Start" and the recording also starts then. Not when he presses on a transportation mode. Therefore we still have "no movements", although they will be significantly less I assume.

**Comment by dimstudio** [comment]
Ok, then what is the difference with the current way? Still not getting it 😕 
Do you think to a weaker or more sporadic recording and storing of sensor data if the user is the transportation mode is not activated?

**Comment by HansBambel** [comment]
Maybe we have the same idea of how the app should be used. This is how I saw it:

**Daniele's use case**
- Start of the day: Start app and recording
- When using any of the transportation modes: use slider to indicate it
- At the end of the day: Stop recording and upload data

**Kevin's use case**
- Start app and recording when about to use transportation
- Stop recording after the end of moving around (user can use multiple transportation methods)
- Reopen when about to use transportation again (gets written to the same files if app not completely closed)
- at the end of day: Upload files

**Differences**
- Daniele: Very big files with likely a lot of unusable data (or mostly "no movement")
- Kevin: Smaller files with recordings being mostly transportation modes

Maybe I misunderstood how you wanted to use the app

**Comment by dimstudio** [comment]
No, your summary is correct. I think your version is more optimal in terms of data but it requires more discipline from the user side. 

In an ideal scenario, this would work, but from personal experience using Strava for running, I realise how difficult it becomes every time to push that recording button. 
I guess we should try it

**Comment by HansBambel** [comment]
Okay, great.

My intention was also that a user may even forget that an app is running in the background of the phone, resulting in him forgetting to turn it on/off.

**Comment by khaleelasyraaf** [comment]
Perhaps a notification to remind the user to turn it off after a few hours of recording?

**Comment by khaleelasyraaf** [comment]
I've added the AlarmManager to make the phone vibrate every two hours after pressing "Start". I couldn't add another notification bar since we already have one for the foreground service (that shows always after pressing "Start"). I think by alerting the user haptically would be better than visually.
